client/my-sites/upgrades/domain-management/transfer/transfer-to-other-site/index.jsx

```diff
@@ -0,0 +1,181 @@
+/**
+ * External Dependencies
+ */
+import React from 'react';
+import { connect } from 'react-redux';
+import {
+	get,
+	omit,
+} from 'lodash';
+import page from 'page';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal Dependencies
+ */
+import Card from 'components/card';
+import SiteSelector from 'components/site-selector';
+import { getCurrentUser } from 'state/current-user/selectors';
+import { getSites } from 'state/selectors';
+import { getSiteBySlug } from 'state/sites/selectors';
+import Header from 'my-sites/upgrades/domain-management/components/header';
+import Main from 'components/main';
+import paths from 'my-sites/upgrades/paths';
+import { getSelectedDomain } from 'lib/domains';
+import NonOwnerCard from 'my-sites/upgrades/domain-management/components/domain/non-owner-card';
+import SectionHeader from 'components/section-header';
+import Dialog from 'components/dialog';
+import { successNotice, errorNotice } from 'state/notices/actions';
+import wp from 'lib/wp';
+
+const wpcom = wp.undocumented();
+
+class TransferToOtherSite extends React.Component {
+
+	static propTypes = {
+		selectedDomainName: React.PropTypes.string.isRequired,
+		selectedSite: React.PropTypes.oneOfType( [
+			React.PropTypes.object,
+			React.PropTypes.bool
+		] ).isRequired,
+		currentUser: React.PropTypes.object.isRequired
+	};
+
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			targetSite: null,
+			showConfirmationDialog: false,
+			disableDialogButtons: false
+		};
+	}
+
+	handleSiteSelect = ( siteSlug ) => {
+		const targetSite = this.props.siteBySlug( siteSlug );
+
+		if ( targetSite ) {
+			this.setState( {
+				targetSite,
+				showConfirmationDialog: true,
+			} );
+		}
+	}
+
+	handleConfirmTransferDomain = ( closeDialog ) => {
+		const { selectedDomainName } = this.props,
+			targetSiteName = this.state.targetSite.name,
+			successMessage = this.props.translate(
+				'%(selectedDomainName)s has been transferred to site: %(targetSiteName)s',
```

Comment:
It's my fault for making those strings so similar, but that's how it is :man_shrugging: